flanker/addresslib/address.py

```diff
@@ -535,7 +535,11 @@ def full_spec(self):
         if not is_pure_ascii(self.mailbox):
```

Comment:
I think `is_pure_ascii` should return `False` if the domain part cannot be IDNA encoded. There should be a contract that if `is_pure_ascii` returns `True` it should be safe to call `EmailAddress.full_spec`.